### PR TITLE
meta/sql: support new backup

### PR DESCRIPTION
**Pull request by jiefenghuang**

### CODE REVIEW

File: pkg/meta/backup.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+	Typ uint32
+	Len uint64
+	Val proto.Message
+}
+
+func (s *BakSegment) String() string {
+	return string(proto.MessageName(s.Val).Name())
+}
+
+func (s *BakSegment) Marshal() ([]byte, error) {
```
Comment (davies) [inline pkg/meta/backup.go:222]: Write the marshaled message into io.Writer directly to avoid memory copy?

File: pkg/meta/pb/backup.proto [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  bytes target = 2;
+}
+
+message SymlinkList {
+  repeated Symlink list = 1;
+}
+
+message Footer {
+  message SegInfo {
+    repeated uint64 offset = 1;
```
Comment (davies) [inline pkg/meta/pb/backup.proto:147]: also type

Comment (davies) [reply pkg/meta/pb/backup.proto:147]: nvm, we have type in infos.

File: pkg/meta/redis_bak.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+//go:build !noredis
+// +build !noredis
+
+/*
+ * JuiceFS, Copyright 2020 Juicedata, Inc.
```
Comment (davies) [inline pkg/meta/redis_bak.go:5]: 2024

File: pkg/meta/backup.go (hunk 1 above)
Comment (jiefenghuang) [reply pkg/meta/backup.go]: done

File: pkg/meta/redis_bak.go (hunk 3 above)
Comment (jiefenghuang) [reply pkg/meta/redis_bak.go]: done

**Pull request merged**